linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/TemplateConfUtils.scala

```diff
@@ -0,0 +1,276 @@
+        varString = s"""\\s*##@set ${confTemplateNameKey}=\\s*.+\\s*"""
+        errString = """\s*##@"""
+      case CodeAndRunTypeUtils.LANGUAGE_TYPE_SCALA =>
+        varString = s"""\\s*///@set ${confTemplateNameKey}=\\s*.+\\s*"""
+        errString = """\s*///@.+"""
+      case _ =>
+        return templateConfName
+    }
+
+    val customRegex = varString.r.unanchored
+    val errRegex = errString.r.unanchored
+    var codeRes = code.replaceAll("\r\n", "\n")
+    // only allow set at fisrt line
+    val res = codeRes.split("\n")
+    if (res.size > 0) {
+      val str = res(0)
+      str match {
+        case customRegex() =>
+          val clearStr = if (str.endsWith(";")) str.substring(0, str.length - 1) else str
+          val res: Array[String] = clearStr.split("=")
+          if (res != null && res.length == 2) {
+            templateConfName = res(1).trim
+            logger.info(s"get template conf name $templateConfName")
+          } else {
+            if (res.length > 2) {
+              throw new LinkisCommonErrorException(
+                20044,
+                s"$str template conf name var defined uncorrectly"
+              )
+            } else {
+              throw new LinkisCommonErrorException(
+                20045,
+                s"template conf name var  was defined uncorrectly:$str"
+              )
+            }
+          }
+        case errRegex() =>
+          logger.warn(
+            s"The template conf name var definition is incorrect:$str,if it is not used, it will not run the error, but it is recommended to use the correct specification to define"
+          )
+        case _ =>
+      }
+    }
+    templateConfName
+  }
+
+  def dealWithTemplateConf(jobRequest: JobRequest, logAppender: lang.StringBuilder): JobRequest = {
+    jobRequest match {
+      case requestPersistTask: JobRequest =>
+        val params = requestPersistTask.getParams
+        val startMap = TaskUtils.getStartupMap(params)
+
+        var templateConflist: util.List[TemplateConfKey] = new util.ArrayList[TemplateConfKey]()
+        var templateName: String = ""
+        // only for Creator:IDE, try to get template conf name from code string. eg:---@set ec.resource.name=xxxx
+        val (user, creator) = LabelUtil.getUserCreator(jobRequest.getLabels)
+        if (EntranceConfiguration.DEFAULT_REQUEST_APPLICATION_NAME.getValue.equals(creator)) {
+          val codeType = LabelUtil.getCodeType(jobRequest.getLabels)
+          templateName =
+            TemplateConfUtils.getCustomTemplateConfName(jobRequest.getExecutionCode, codeType)
+        }
+
+        // code template name > start params template uuid
+        if (StringUtils.isBlank(templateName)) {
+          logger.debug("jobRequest startMap param template name is empty")
+
+          logger.info("jobRequest startMap params :{} ", startMap)
+          val templateUuid = startMap.getOrDefault(LabelKeyConstant.TEMPLATE_CONF_KEY, "").toString
```

Comment:
entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/TemplateConfUtils.scala:204: value TEMPLATE_CONF_KEY is not a member of object org.apache.linkis.manager.label.constant.LabelKeyConstant